qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -331,53 +331,46 @@ def _format_results(output):
         """Assemble one or more Qobj for running on the simulator"""
-        # Remove quantum error instructions from circuits and add to
-        # run option noise model
-        circuits, optypes, run_options = self._assemble_noise_model(circuits, **run_options)
-
-        # This conditional check can be removed when we remove passing
-        # qobj to run
         if isinstance(circuits, (QasmQobj, PulseQobj)):
             qobj = circuits
-        elif parameter_binds:
-            # Handle parameter binding
-            parameterizations = self._convert_binds(circuits, parameter_binds)
-            assemble_binds = []
-            assemble_binds.append({param: 1 for bind in parameter_binds for param in bind})
-
-            qobj = assemble(
-                compile_circuit(
-                    circuits,
-                    basis_gates=self.configuration().basis_gates,
-                    optypes=optypes),
-                self,
-                parameter_binds=assemble_binds,
-                parameterizations=parameterizations)
         else:
-            qobj = assemble(
-                compile_circuit(
+            # Generate optypes for circuit
+            # Generate opsets of instructions
+            if not isinstance(circuits, list):
+                circuits = [circuits]
+            optypes = [circuit_optypes(circ) for circ in circuits]
+
+            # Compile Qasm3 instructions
+            circuits, optypes = compile_circuit(
+                circuits,
+                basis_gates=self.configuration().basis_gates,
+                optypes=optypes)
+
+            # run option noise model
+            circuits, optypes, run_options = self._assemble_noise_model(
+                circuits, optypes, **run_options)
+
+            if parameter_binds:
+                # Handle parameter binding
+                parameterizations = self._convert_binds(circuits, parameter_binds)
+                assemble_binds = []
+                assemble_binds.append({param: 1 for bind in parameter_binds for param in bind})
+
+                qobj = assemble(
                     circuits,
-                    basis_gates=self.configuration().basis_gates,
-                    optypes=optypes),
-                self)
-
-        # Add optypes to qobj
-        # We convert to strings to avoid pybinding of types
-        qobj.config.optypes = [
-            [i.__name__ for i in optype] if optype else [] for optype in optypes
-        ]
+                    backend=self,
+                    parameter_binds=assemble_binds,
+                    parameterizations=parameterizations)
+            else:
+                qobj = assemble(circuits, backend=self)
+
+            # Add optypes to qobj
+            # We convert to strings to avoid pybinding of types
+            qobj.config.optypes = [
+                [i.__name__ for i in optype] if optype else []
+                for optype in optypes]
```

Comment:
Gross, but there's not really an alternative in this case, I think.  Would it be better to pass this as a `set` still - I don't know how that's handled in C++, though.

Reply:
There isnt really a JSON equivalent for set, so its easier to just convert to a list.

Reply:
Well, it seemed to work changing to set, so i guess thats ok.